core/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -664,6 +664,20 @@ public MappedFieldType fullName(String fullName) {
     }
 
     /**
+     * Returns a {@link MappedFieldType} that is nested under the field for the give fullName.
+     *
+     * If multiple nested types have fields with the same full name, the first is returned.
+     */
+    public MappedFieldType subfieldType(String fullName, Class<? extends MappedFieldType> mappedFieldType) {
+        for (String fieldName : fieldTypes.simpleMatchToFullName(fullName + ".*")) {
+            if (mappedFieldType.isInstance(fieldTypes.get(fieldName))) {
+                return fieldTypes.get(fieldName);
+            }
+        }
+        return null;
```

Comment:
Again, it would be better if this was to return an optional, but given the time that you have left until submission, this works.